woocommerce/woocommerce-gateway-stripe
# Tweaks in the connect account modal

Closes #2188

## Changes proposed in this Pull Request:

- Disable autocomplete for input fields of the connect modal.
- Return feedback to the user upon a failed connection with the keys entered.

## Testing instructions

<img src="https://user-images.githubusercontent.com/10233985/141596690-0f6684c4-902b-47bc-8439-d416920d8f12.png" width="600" />

<img src="https://user-images.githubusercontent.com/10233985/143306702-b883b236-489f-491e-a210-5845b04931ce.png" width="600" />

From [the stripe settings dashboard](http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe&panel=settings):

1 - Click `Edit account keys`

2 - Check that all input fields don't show autocomplete options. For live and test keys.

3 - Enter valid live keys and save.

Expected Result: 
- Save button is blocked for a while. 
- Modal closes automatically. 
- Message `Account keys Saved` is shown. 
- Page is not reloaded. 

4 - Enter valid test keys and save.

Expected Result: 
- Save button is blocked for a while. 
- Modal closes automatically. 
- Message `Account keys Saved` is shown. 
- Page is not reloaded. 

5 - Enter invalid live keys and save.

Expected Result: 
- Save button is blocked for a while. 
- Modal **does not close** automatically. 
- Message `Error saving account keys.` is shown. 
- Page is not reloaded. 

6 - Enter invalid test keys and save.

Expected Result: 
- Save button is blocked for a while. 
- Modal **does not close** automatically. 
- Message `Error saving account keys.` is shown. 
- Page is not reloaded. 

---


**Post merge**
